src/Illuminate/Collections/helpers.php

```diff
@@ -19,6 +19,30 @@ function collect($value = [])
     }
 }
 
+if (! function_exists('deepCollect')) {
+    /**
+     * Recursively convert an array and its nested arrays into Laravel collections.
+     *
+     * This function takes an array (or any iterable) and transforms it into an instance
+     * of Laravel's Collection class. It applies the transformation to all nested arrays,
+     * ensuring that every level of the array structure is converted into collections.
+     *
+     * @param  mixed  $value  The input value, which can be an array, an iterable, or any other data type.
+     * @return mixed If the value is an array, a Collection instance is returned with all nested arrays
+     *               converted. If the value is not an array, it is returned as-is.
+     */
+    function deepCollect($value)
+    {
+        // If the value is an array, convert it into a Collection and apply deepCollect recursively
+        if (is_array($value)) {
+            return collect($value)->map(fn ($item) => deepCollect($item));
```

Comment:
You might be able to use first-class callable syntax here:
```suggestion
            return collect($value)->map(deepCollect(...));
```